codebuild/bin/install_s2n_head.sh

```diff
@@ -12,31 +12,50 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-set -ex
-pushd "$(pwd)"
+set -e
 
 usage() {
     echo "install_s2n_head.sh build_dir"
     exit 1
 }
 
+BUILD_DIR=$1
+SRC_ROOT=${SRC_ROOT:-$(pwd)}
+
 if [ "$#" -ne "1" ]; then
     usage
 fi
 
-BUILD_DIR=$1
-source codebuild/bin/jobs.sh
-cd "$BUILD_DIR"
-
-# Clone the most recent s2n commit
-git clone --depth=1 https://github.com/aws/s2n-tls s2n_head
-cmake ./s2n_head -Bbuild -DCMAKE_PREFIX_PATH="$LIBCRYPTO_ROOT" -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=on -DBUILD_TESTING=on
-cmake --build ./build -- -j $JOBS
-
-# Copy new executables to bin directory
-cp -f "$BUILD_DIR"/build/bin/s2nc "$BASE_S2N_DIR"/bin/s2nc_head
-cp -f "$BUILD_DIR"/build/bin/s2nd "$BASE_S2N_DIR"/bin/s2nd_head
+# CMake(nix) and Make are using different directory structures.
+if [[ "$IN_NIX_SHELL" ]]; then
+    export DEST_DIR="$SRC_ROOT"/build/bin
+    # Safety measure
+    mkdir -p "$DEST_DIR"
+else
+    export DEST_DIR="$SRC_ROOT"/bin
+fi
 
-popd
+if [[ ! -x "$DEST_DIR/s2nc_head" ]]; then
+    if [[ ! -d "s2n_head" ]]; then
+        # Clone the most recent s2n commit
+        git clone --branch main --single-branch . s2n_head
+    else
+        cd s2n_head
+        echo "Checking the age of s2n_head..."
+        test $(date -d '-3 days' +%s) -lt $(git log -1 --format="%at") || echo "s2n_head is too old, refusing to use it";exit 1
+        cd ..
+    fi
```

Comment:
I don't think this is right. You're only checking the age of s2n_head if s2nc_head doesn't exist, so you're not going to rebuild an old s2nc_head. We also shouldn't need to just die, we could just rebuild.

I think your logic needs to be something like:
```
if [[ -d "s2n_head" ]]; then
    now=$(date +%s)
    last_modified=$(stat -c %Y s2n_head)
    days_old=$(( ($now - $last_modified) / 86400))
    if ((days_old > 1)); then
        rm -rf s2n_head
    fi
fi

if [[ -d "s2n_head" ]]; then
    echo "s2n_head already exists: $days_old days old"
else
    // do all your install work
fi
```

Reply:
Agreed- this is more straightforward; updated.